metabase/metabase
# First take at parameterized data app page titles

Adds basic implementation of parameterized page titles.

The implementation is based on the approach taken in the #23533 prototype PR. The key difference is that this PR only lets us parameterize page titles as opposed to markdown cards. Also adds some UI to explore available data a user can mention in the page title.

The resulting title is stored in the page's nav item. That required changes to the dashboard code to make it more aware of data apps and pages and added a bit of extra diff.

### To Verify

1. New > App > Sample Dataset > People > Create
2. Open a user detail page
3. Make sure page name is used by default when there is no title template provided yet (should say something like People Detail)
5. Enter editing mode
6. Clear the name input and make sure the "Save" button becomes disabled
7. Use the "data" icon next to the name input to browse data you can put into the page title
8. Build a title template that looks good to you, click Save
9. Go through a few pages and make sure you see expected data in page titles

#### Demo

https://user-images.githubusercontent.com/17258145/195879433-315ebbeb-79c6-41f5-9284-ddc0c982d2f4.mp4
